### PR TITLE
Enable more legacy tests

### PR DESCRIPTION
**Pull request by jfversluis**
### Description of Change

Ignore the failing iOS tests for now on the legacy UI tests so that we can get the remaining tests going in an automated fashion.

### Related

https://github.com/dotnet/maui/pull/17971

### CODE REVIEW

**Review by PureWeen — CHANGES_REQUESTED** [review]
iOS tests seem like they didn't run
https://dev.azure.com/xamarin/public/_build/results?buildId=98547&view=results

File: eng/devices/devices-shared.cake [hunk 1]
```diff
@@ -8,3 +8,15 @@ if (string.Equals(TARGET, "uitest", StringComparison.OrdinalIgnoreCase))
...
+void FailRunOnOnlyInconclusiveTests(string testResultsFile)
+{
+    // When all tests are inconclusive the run does not fail, check if this is the case and fail the pipeline so we get notified
+	var totalTestCount = XmlPeek(testResultsFile, "/test-run/@total");
+	var inconclusiveTestCount = XmlPeek(testResultsFile, "/test-run/@inconclusive");
+	
+	if (totalTestCount.Equals(inconclusiveTestCount))
+    {
+		throw new Exception("All tests are marked inconclusive, no tests ran. There is probably something wrong with running the tests.");
+	}
```
Comment (jfversluis) [inline eng/devices/devices-shared.cake:21]: We had a case where we updated the UITest dependency and all iOS tests came up as inconclusive. For NUnit this still passes and so we thought: hooray, all green!

To make sure we notice something might be up, check the results and fail the test run if this scenario comes up.

Comment (jfversluis) [reply eng/devices/devices-shared.cake]: Moved this into a separate PR #17971

**Comment by jfversluis** [comment]
@PureWeen as discussed, this PR as-is, moved the check for the tests not running to a separate PR here: #17971

**Review by PureWeen — APPROVED** [review]

**Pull request merged**